### DISCUSSION TITLE
After restarting container can't access the login page and back to Basic OpenAM Setup page

### DISCUSSION THREAD

**Discussion by Old-Kevin** [OpenIdentityPlatform/OpenAM#399] [Q&A]
I setup Basic OpenAM and log into the console of openam.
But After I restarted the container, I can not log into the console of openam.
`docker container stop openam`
`docker container start openam`

It redirects to the Basic OpenAM Configuration page.
Who can help me with this?

**Comment by maximthomas** [comment]
You should mount OpenAM configuration volume to your host machine. OpenAM configuration volume defined in `OPENAM_DATA_DIR` environment variable with default value `"/usr/openam/config"`
https://github.com/OpenIdentityPlatform/OpenAM/blob/master/openam-distribution/openam-distribution-docker/Dockerfile#L7

**Reply by Old-Kevin** [reply]
Thank you for your reply.
I tried as you mentioned but it did not work. Still got the same result.
What I tried is that I run the docker by the following command.
docker run -d -h openam.example.com -v /root/openam:/usr/openam/config -p 8443:8080 --name openam openidentityplatform/openam

**Comment by maximthomas** [comment]
Do you have any data in the mounted volume?

**Reply by Old-Kevin** [reply]
No. I created a new directory, so the directory /root/openam is empty.

**Reply by maximthomas** [reply]
You should not create a new directory. Docker will create it itself

**Reply by Old-Kevin** [reply]
I deleted the directory I created and tried again. 
Still got the same result.
There is no configuration file in both directories.
+ /root/openam
+ /usr/openam/config

It seems the issue is not at the mount volume.
If I just restart the container, the data in the container will be stored continuously.
The problem here is there are no configuration file stores in the config directory.

**Reply by maximthomas** [reply]
Check contens /usr/openam/config directory in your docker container, and in mounted volume. Seems there is an issue with permissions.

**Reply by Old-Kevin** [reply]
![image](https://user-images.githubusercontent.com/56917775/133882750-0c84b64d-7258-4c2f-839e-1edb048915a8.png)
When running a container without specifying a user, I think the container will uses root by default.
The screenshot above is the directory inside the container.
It seems the permissions of the directory are enough.

**Reply by maximthomas** [reply]
What OS are you using? Does docker have rights to write in your host machine /root/openam direcotry?

**Reply by Old-Kevin** [reply]
I am using CentOS Linux release 7.9.2009 (Core).
Docker container will use root by default so yes, docker has rights to write the directory.

**Reply by Old-Kevin** [reply]
I checked the image and found that the value of OPENAM_DATA_DIR didn't pass to the CATALINA_OPTS.
I don't know the cause yet.

            "Env": [
                "PATH=/usr/local/tomcat/bin:/usr/local/tomcat/bin:/usr/local/openjdk-8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "JAVA_HOME=/usr/local/openjdk-8",
                "LANG=C.UTF-8",
                "JAVA_VERSION=8u302",
                "CATALINA_HOME=/usr/local/tomcat",
                "TOMCAT_NATIVE_LIBDIR=/usr/local/tomcat/native-jni-lib",
                "LD_LIBRARY_PATH=/usr/local/tomcat/native-jni-lib",
                "GPG_KEYS=05AB33110949707C93A279E3D3EFE6B686867BA6 07E48665A34DCAFAE522E5E6266191C37C037D42 47309207D818FFD8DCD3F83F1931D684307A10A5 541FBE7D8F78B25E055DDEE13C370389288584E7 5C3C5F3E314C866292F359A8F3AD5C94A67F707E 765908099ACF92702C7D949BFA0C35EA8AA299F1 79F7026C690BAA50B92CD8B66A3AD3F4F22C4FED 9BA44C2621385CB966EBA586F72C284D731FABEE A27677289986DB50844682F8ACB77FC2E86E29AC A9C5DF4D22E99998D9875A5110C01C5A2F6059E7 DCFD35E0BF8CA7344752DE8B6FB21E8933C60243 F3A04C595DB5B6A5F1ECA43E3B7BBB100D811BBE F7DA48BB64BCB84ECBA7EE6935CD23C10D498E23",
                "TOMCAT_MAJOR=8",
                "TOMCAT_VERSION=8.5.71",
                "TOMCAT_SHA512=292a3f856b0a8c1d11fd1ba252cabd94794201cda4f951dd0522764449bed90f2f43a4a667cd6d28ce13c3b2096736978d9df91709c168ba7133c51544446433",
                "OPENAM_USER=openam",
                "OPENAM_DATA_DIR=/usr/openam/config",
                "CATALINA_OPTS=-Xmx2048m -server -Dcom.iplanet.services.configpath= -Dcom.sun.identity.configuration.directory="
            ],

**Reply by vharseko** [reply]
@maximthomas OPENAM_DATA_DIR didn't pass to the CATALINA_OPTS

**Reply by maximthomas** [reply]
The issue was fixed in the latest Dockerfile:
https://github.com/OpenIdentityPlatform/OpenAM/blob/6e1fd65f578e04fc06eef18ccac1f17fbb961ab0/openam-distribution/openam-distribution-docker/Dockerfile#L5-L9

**Reply by vkumar535** [reply]
I have updated dockerfile with shared line and it worked

**Comment by vkumar535** [comment]
I have also set up the same and facing same issue, any solution or suggestion to fix this?

**Reply by Old-Kevin** [reply]
I downloaded Dockerfile and changed the following lines - add ENV to CATALINA_OPTS Line
Built an image using the changed dockerfile and It worked.

`ENV CATALINA_HOME=/usr/local/tomcat \
    PATH=$CATALINA_HOME/bin:$PATH \
    OPENAM_USER="openam" \
    OPENAM_DATA_DIR=/usr/openam/config
    ENV CATALINA_OPTS="-Xmx2048m -server -Dcom.iplanet.services.configpath=${OPENAM_DATA_DIR} -Dcom.sun.identity.configuration.directory=${OPENAM_DATA_DIR}"`
